controllers/auth.js

```diff
@@ -0,0 +1,33 @@
+const User = require('../models/User');
+const { StatusCodes } = require('http-status-codes');
+const { BadRequestError, UnauthenticatedError } = require('../errors');
+
+const register = async (req, res) => {
+  const user = await User.create({ ...req.body });
+  const token = user.createJWT();
+  res.status(StatusCodes.CREATED).json({ user: { name: user.name }, token });
+};
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+
+  if (!email || !password) {
+    throw new BadRequestError('Please provide email and password');
+  }
+  const user = await User.findOne({ email });
+  const isPasswordCorrect = await user.comparePassword(password);
+  if (!isPasswordCorrect) {
+    throw new UnauthenticatedError('Invalid Credentials');
+  }
+  // compare password
+  if (!user) {
```

Comment:
this line of code that checks if there's no user found, should come directly after the line of code that searches our DB for the user. If there's no user, there's no reason to execute the code that compares the password, since the user doesn't exist

Reply:
Thank you for the feedback - that's very good to know! This code was from the instructor's assignment and his own code. I think he put this to demonstrate how we can either make custom error messages or generic error messages.

Reply:
Ok, I've changed it! Thank you.